src/Microsoft.TestPlatform.PlatformAbstractions/common/System/ProcessHelper.cs

```diff
@@ -40,6 +40,13 @@ internal ProcessHelper(IEnvironment environment)
         _environment = environment;
     }
 
+    /// <summary>
+    /// Gets or sets the set of environment variables to be used when spawning a new process.
+    /// Should this set of environment variables be null, the environment variables inherited from
+    /// the parent process will be used.
+    /// </summary>
+    public static IDictionary<string, string?>? ExternalEnvironmentVariables { get; set; }
```

Comment:
```suggestion
    public static IDictionary<string, string?>? DefaultEnvironmentVariables { get; set; }
```

Reply:
I like the original name better, because this is not the default.

Reply:
I also prefer the original name. Maybe we could go with `ProcessEnvironmentVariables` or `BaselineEnvironmentVariables`?

Reply:
I have indented this prop like "you inherit from parent and plus you'll always add this default list" so "external" is not super clear to me, as a static "always present" environment variables. But I'm fine with both.

Reply:
I think I may have partially addressed this naming issue by fixing another comment I got from Jakub. I've added an extra property called `InheritEnvironmentVariables` to be used in tandem with `ExternalEnvironmentVariables`. If `InheritEnvironmentVariables` is set to true, then those env variables are just extra variables to be added to the ones we've already inherited from the parent process, otherwise we first clean up the baseline and add only our external environment variables afterwards.